src/main/java/org/broadinstitute/hellbender/tools/exome/segmentation/JointAFCRSegmenter.java

```diff
@@ -0,0 +1,196 @@
+
+    private JointAFCRSegmenter(final List<SimpleInterval> positions,
+                              final List<TargetOrHetDatum> data,
+                              final List<AFCRHiddenState> hiddenStateValues,
+                              final double[] weights,
+                              final double concentration,
+                              final double memoryLength,
+                              final AlleleFractionGlobalParameters parameters,
+                              final AllelicPanelOfNormals allelicPoN,
+                              final double log2CoverageCauchyWidth) {
+
+        super(positions, data, hiddenStateValues, weights, concentration, memoryLength);
+        this.parameters = parameters;
+        this.allelicPoN = allelicPoN;
+        this.log2CoverageCauchyWidth = log2CoverageCauchyWidth;
+        logStatesAndWeights();
+    }
+
+    private void logStatesAndWeights() {
+        final StringBuilder message = new StringBuilder("Joint segmenter has following (copy-ratio, allele-fraction): weight pairs: ");
+        for (int n = 0; n < weights.length; n++) {
+            final AFCRHiddenState state = hiddenStateValues.get(n);
+            final double cr = state.getLog2CopyRatio();
+            final double af = state.getMinorAlleleFraction();
+            final double weight = weights[n];
+            message.append(String.format("(%f, %f): %f", cr, af, weight) + ((n < weights.length - 1) ? ";" : "."));
+        }
+        logger.info(message);
+    }
+
+    public static JointAFCRSegmenter createJointSegmenter(final CopyRatioSegmenter copyRatioSegmenter, final AlleleFractionSegmenter alleleFractionSegmenter) {
+        copyRatioSegmenter.makeSureParametersHaveBeenLearned();
+        alleleFractionSegmenter.makeSureParametersHaveBeenLearned();
+        List<IntervalDatumIndex> targetsWithCopyRatio = IntStream.range(0, copyRatioSegmenter.numPositions())
+                .mapToObj(n -> new IntervalDatumIndex(copyRatioSegmenter.getPosition(n), new TargetOrHetDatum(copyRatioSegmenter.getDatum(n)), n))
+                .collect(Collectors.toList());
+        List<IntervalDatumIndex> hetsWithAllelicCount = IntStream.range(0, alleleFractionSegmenter.numPositions())
+                .mapToObj(n -> new IntervalDatumIndex(alleleFractionSegmenter.getPosition(n), new TargetOrHetDatum(alleleFractionSegmenter.getDatum(n)), n))
+                .collect(Collectors.toList());
+        List<IntervalDatumIndex> positionsWithData = CollectionUtils.collate(targetsWithCopyRatio, hetsWithAllelicCount);
+
+        final List<SimpleInterval> allPositions = positionsWithData.stream().map(IntervalDatumIndex::getInterval).collect(Collectors.toList());
+        final List<TargetOrHetDatum> allData = positionsWithData.stream().map(IntervalDatumIndex::getDatum).collect(Collectors.toList());
+
+
+        // pairs of targets with immediately preceding and following hets, and vice versa
```

Comment:
This comment is a bit inexact, and I had to look at the code to understand its meaning...can you be a little more verbose?